src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -2009,7 +2009,7 @@ private BoundExpression BindSuppressNullableWarningExpression(PostfixUnaryExpres
             if ((object)type != null)
             {
                 // Report diagnostic if the type consists of no reference types.
-                if ((object)type.VisitType((t, a, c) => t.IsErrorType() || t.IsReferenceType, (object)null, canDigThroughNullable: true) == null)
+                if ((object)type.VisitType((t, a, c) => t.IsErrorType() || t.IsReferenceType || t.IsUnconstrainedTypeParameter(), (object)null, canDigThroughNullable: true) == null)
```

Comment:
Not related to this PR: might it make sense to move this diagnostic to the nullable walker?

Reply:
Yes. Included that in comment.

---
In reply to: [177565882](https://github.com/dotnet/roslyn/pull/25733#discussion_r177565882) [](ancestors = 177565882)